packages/a11y-base/src/focus-restoration-controller.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { getDeepActiveElement } from './focus-utils.js';
+
+/**
+ * A controller for saving a focused node and restoring focus to it later.
+ */
+export class FocusRestorationController {
+  /**
+   * Saves the given node as a target for restoring focus to
+   * when `restoreFocus()` is called. If no node is provided,
+   * the currently focused node in the DOM is saved as a target.
+   *
+   * @param {Node | null | undefined} focusNode
+   */
+  saveFocus(focusNode) {
+    this.focusNode = focusNode || getDeepActiveElement();
+  }
+
+  /**
+   * Restores focus to the target node that was saved previously with `saveFocus()`.
+   */
+  restoreFocus() {
```

Comment:
Question: the original logic to restore focus also cleared the `focusNode` element reference, but here it doesn't. There's a slightly different behavior, so I was wondering if this is intentional. If so, maybe it is worth adding a test to check this scenario.

Reply:
Well, I thought it's not so important because I couldn't think of a valid use-case where you'd need to call `saveFocus` and then `restoreFocus` multiple times. However, now I think that I'll revert this line and add a test as you suggested.

Reply:
Updated.